data/shared/src/main/scala/sigma/data/ExactNumeric.scala

```diff
@@ -30,6 +32,66 @@ trait ExactNumeric[T] {
   def toInt(x: T): Int = n.toInt(x)
   def toLong(x: T): Long = n.toLong(x)
 
+  /** Returns a big-endian representation of this value in a collection of bytes.
+    * For example, the `Int` value `0x12131415` would yield the
+    * collection of bytes [0x12, 0x13, 0x14, 0x15]
+    */
+  def toBigEndianBytes(x: T): Coll[Byte]
+
+  /**
+    * Returns a big-endian binary representation of this value as boolean array.
+    */
+  def toBits(x: T): Coll[Boolean] = {
+
+    def isBitSet(byte: Byte)(bit: Int): Boolean = ((byte >> bit) & 1) == 1
+
+    def byte2Bools(b: Byte): Array[Boolean] =  (0 to 7).toArray.reverse.map(isBitSet(b))
```

Comment:
this methods creates 1 Range allocation + 3 allocations of arrays.

Reply:
optimized